R/btw_client.R

```diff
@@ -519,6 +532,44 @@ btw_client_config <- function(client = NULL, tools = NULL, config = list()) {
   config
 }
 
+flatten_and_check_tools <- function(tools) {
+  if (isFALSE(tools)) {
+    return(list())
+  }
+
+  if (inherits(tools, "ellmer::ToolDef")) {
+    tools <- list(tools)
+  }
+
+  if (is.character(tools)) {
+    tools <- btw_tools(tools)
+    return(tools)
+  }
+
+  if (!is.list(tools)) {
+    cli::cli_abort(c(
+      "Invalid {.arg tools}: Must be a character vector of {.pkg btw} tool names, or list of Tool objects and {.pkg btw} tool names.",
+      i = "See {.help btw::btw_tools()} for more information about available tools and names."
+    ))
+  }
+
+  flat_tools <- list()
+  for (i in seq_along(tools)) {
+    tool <- tools[[i]]
+    if (inherits(tool, "ellmer::ToolDef")) {
+      flat_tools <- c(flat_tools, list(tool))
+    } else if (is.character(tool)) {
+      flat_tools <- c(flat_tools, btw_tools(tool))
+    } else {
+      cli::cli_abort(
+        "Invalid tool in {.arg tools[[{i}]]}: Must be a character vector or Tool object, not {.obj_type_friendly {tools[[i]]}}."
```

Comment:
Noting the substituion here--is the intent that the user sees e.g. `tools[[3]]`? Fine if so, but caught my eye.